docs/tutorial/using-checkbox/basic-launcher.rst

```diff
@@ -0,0 +1,260 @@
+.. _base_tutorial_launcher:
+
+=========
+Launchers
+=========
+
+Checkbox features and behaviors can be configured and customized using
+a launcher.
+
+In this section, you will use a launcher to filter the list of test plans
+displayed when running Checkbox, pre-select a test plan, automatically execute
+a test plan without manual interaction and configure the environment variables
+used by some of the test cases. Finally, you will tailor the output generated
+by Checkbox to your needs.
+
+Filter the list of test plans
+=============================
+
+When you ran Checkbox in the previous section, you were probably overwhelmed
+with the number of test plans available. Using a launcher, it is possible to
+filter out the list of test plans to only select the ones that matter to you.
+
+Create a file named ``mylauncher`` and add the following information in it:
+
+.. code-block:: none
+    :caption: mylauncher
+    :name: initial-launcher
+
+    [launcher]
+    launcher_version = 1
+    app_id = com.canonical.certification:tutorial
+
+    [test plan]
+    filter = *TODO*
+
+Save it, then launch Checkbox using this launcher:
+
+.. code-block:: none
+
+    checkbox.checkbox-cli launcher mylauncher
```

Comment:
I had actually never thought about the shebang not pointing to it.

But when I run `checkbox.checkbox-cli --help` I get:

```
usage: checkbox-cli [-h] [-v] [--debug] [--clear-cache] [--clear-old-sessions] [--version]
                    {check-config,launcher,list,run,startprovider,submit,show,list-bootstrapped,merge-reports,merge-submissions,tp-export,service,remote}
```

and then

```
checkbox.checkbox-cli launcher --help
usage: checkbox-cli [-h] [--verify] [--title SESSION_NAME] [-m MESSAGE] [--dont-suppress-output] [-v] [--debug] [--clear-cache] [--clear-old-sessions] [--version] [LAUNCHER]

positional arguments:
  LAUNCHER              launcher definition file to use

options:
  -h, --help            show this help message and exit
  --verify              only validate the launcher
  --title SESSION_NAME  title of the session to use
  -m MESSAGE, --message MESSAGE
                        submission description
  --dont-suppress-output
                        Absolutely always show command output
  -v, --verbose         print more logging from checkbox
  --debug               print debug messages from checkbox
  --clear-cache         remove cached results from the system
  --clear-old-sessions  remove previous sessions' data
  --version             show program's version information and exit
```

What I mean is that in the default help, there is no mention of a optional `[LAUNCHER]` right after the `checkbox-cli` command. Is it something your PR (#567) changes?

Reply:
if you look below the first command you have a list of subcommands, checkbox always assumes that you want launcher if you did not ask for help, I have made it a bit better in #567 but this is the behaviour. We have a default command (right now it is launcher) if you dont ask for anything it will be assumed